### ISSUE TITLE
index subset not catching more than one method error

### ISSUE THREAD

**Issue by j-harbin** [ArgoCanada/argoFloats#254]
Hi Team,

Working on https://github.com/ArgoCanada/argoFloats/issues/253 presented by Dan, and I noticed when we do the following:

```
library(argoFloats)
data("index")
index1 <- subset(index, dataMode='delayed', institution='AO')
```
We receive
```
Kept 517 profiles (52.9%)
```
However, doing
```
index1[['institution']]
```
demonstrates that the subset didn't capture the subset by institution. This is expected because we have only set up to be able to do one subset at a time, however, for some reason it did not pick up the warning message
"in subset,argoFloats-method() :\n  cannot give more than one method in the '...' argument"

I will be looking into this tomorrow.

**Comment by dankelley** [comment]
The second test, `index1[['institution']]`, is not calling `subset()`; rather, it's calling `[[`.  That's why we don't get that error message.

This is confusing, because in other aspects of R, `subset()` and `[[` are (nearly) equivalent, and so the word "subset" is sometimes used for `[[`.  Think of it this way, though:
* `subset(x)` returns a portion of `x`, and it has the same *nature* as x.  If `x` is a data frame, then the returned result will be a data frame.
* `x[["something"]]` returns only this "something" from x, and it does not have the same *nature* as x.  If `x` is a data frame, then "something" might be indicating a column to select.

**Comment by dankelley** [comment]
Oh, sorry, I think I misread your comment.  What you're really saying is that `subset()` ought to notice and report the fact that it was called with multiple criteria.  You can ignore my last comment :-)

**Comment by j-harbin** [comment]
Another one letter bug! This issue has been fixed in commit 4255fda91dc64da20933faca016cce72c781e6bb of the develop branch.

**Comment by dankelley** [comment]
Let's invent an acronym: OLB.
